Build failure: python312Packages.radio-beam

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix-build -A python312Packages.radio-beam`

### Build log

```console
Running phase: pytestCheckPhase
Executing pytestCheckPhase
ImportError while loading conftest '/private/tmp/nix-build-python3.12-radio-beam-0.3.7.drv-0/radio-beam-0.3.7/radio_beam/conftest.py'.
radio_beam/conftest.py:7: in <module>
    from distutils.version import LooseVersion
E   ModuleNotFoundError: No module named 'distutils'
```

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.4.0, macOS 14.4`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.21.0`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/radio-beam/default.nix
+++ b/pkgs/development/python-modules/radio-beam/default.nix
@@ -1,5 +1,6 @@
 { lib
 , fetchPypi
+, fetchpatch2
 , buildPythonPackage
 , setuptools-scm
 , astropy
@@ -20,6 +21,15 @@ buildPythonPackage rec {
     inherit pname version;
     hash = "sha256-7AFkuuYLzibwwgz6zrFw0fBXCnGLzdm4OgT+Chve5jU=";
   };
+
+  # Fix distutils deprecation in Python 3.12. See:
+  # https://github.com/radio-astro-tools/radio-beam/pull/124
+  patches = [
+    (fetchpatch2 {
+      url = "https://github.com/radio-astro-tools/radio-beam/commit/1eb0216c8d7f5a4494d8d1fe8c79b48425a9c491.patch";
+      hash = "sha256-kTJF/cnkJCjJI2psvs+4MWFn/+b8TvUWjdfYu5ot0XU=";
+    })
+  ];
 
   nativeBuildInputs = [
     setuptools-scm
@@ -50,5 +60,3 @@ buildPythonPackage rec {
     maintainers = with maintainers; [ smaret ];
   };
 }
-
-

